### DISCUSSION TITLE
Plugin scripting reference

### DISCUSSION THREAD

**Discussion by dvorgaz** [WhiteMagic/JoystickGremlin#569] [Q&A]
Hello, is there a full api reference for writing plugins? For example I want to get the current active mode but I don't know if there's such a function. From the example plugins I know gremlin.control_action.switch_mode() can be used to set the mode, but I have no idea what other functions are in there. Looking through the source I couldn't find what gremlin.control_action contains or what it is.

**Comment by WhiteMagic** [comment]
No there isn't and the R13 code base is also not really well suited for Sphinx to extract a decent API. The thing you're after though can be found in `gremlin.event_handler.EventHandler.active_mode` and mode changes are communicated via a Qt signal.
